plugins/image-snapshot/src/main.ts

```diff
@@ -0,0 +1,160 @@
+    failureThreshold: 0.01,
+    blur: 1
+  };
+
+  constructor(options : ImageSnapshotPluginOptions = { globalMatchOptions: {} }) {
+    this.getSnapshotsDir = options.getSnapshotsDir ?? this.getSnapshotsDir;
+    this.imageWidth = options.imageWidth ?? this.imageWidth;
+    this.imageHeight = options.imageHeight ?? this.imageHeight;
+    this.globalMatchOptions = {
+      ...this.globalMatchOptions,
+      ...options.globalMatchOptions
+    };
+  }
+
+  apply(proof: Proof) {
+    const _globalMatchOptions = this.globalMatchOptions;
+    const _getSnapshotsDir = this.getSnapshotsDir;
+    const _imageWidth = this.imageWidth;
+    const _imageHeight = this.imageHeight;
+    const snapshotState = new SnapshotState('', {
+      updateSnapshot: this.updateSnapshots ? "all" : "new",
+      getPrettier: () => null,
+      getBabelTraverse: () => traverse
+    });;
+
+    proof.hooks.start.tap('image-snapshot', () => {
+      if (!fs.existsSync(SCREEN_SHOT_DIR)) {
+        fs.mkdirSync(SCREEN_SHOT_DIR);
+      }
+    });
+
+    proof.hooks.testStart.tap('image-snapshot', (t: ProofTest) => {
+      t.hooks.beforeExecute.tapPromise(
+        'image-snapshot',
+        async (_testFunc: any, testArgs: TestHookArgs) => {
+          testArgs.browser.addCommand(
+            'matchImageSnapshot',
+            async function(this: ImageSnapshotBrowser, {
+              windowWidth = 1280,
+              windowHeight = 800,
+              ...rest
+            } : ImageSnapshotOptions = {}) {
+              await this.setWindowSize(windowWidth, windowHeight);
+              
+              const testName = `${kebabCase(testArgs.config.kind)}--${kebabCase(testArgs.config.story)}`;
+              
+              const screenShotBuffer = await this.saveScreenshot(`${SCREEN_SHOT_DIR}/${testName}.png`);
+              
+              const resized = await sharp(screenShotBuffer)
+                .resize(_imageWidth, _imageHeight)
+                .png()
+                .toBuffer();
+
+              const snapshotDir = _getSnapshotsDir({
+                kind: testArgs.config.kind,
+                story: testArgs.config.story
+              });
+
+              const result = (toMatchImageSnapshot as ToMatchImageSnapshotFunc).apply({
+                snapshotState,
+                isNot: false,
+                testPath: snapshotDir,
+                currentTestName: testName
+              }, [resized, {
+                customSnapshotsDir: snapshotDir,
+                allowSizeMismatch: true,
+                ..._globalMatchOptions,
+                ...rest
+              }]);
+
+              if (!result.pass) {
+                throw new Error(result.message());
+              }
+
+              return result.pass;
+            }
+          )
+        }
+      );
+    });
+
+    proof.hooks.end.tapPromise('image-snapshot', async () => {
+      if (fs.existsSync(SCREEN_SHOT_DIR)) {
+        fs.rmdirSync(SCREEN_SHOT_DIR, { recursive: true });
```

Comment:
This cleanup doesn't seem to be called if a test throws an error.

Reply:
That's a little surprising -- the `end` hook _should_ include the summary of all of the tests ran, including any failed ones (ones that threw exceptions during their run). I believe we use this same hook in some other plugins (`junit`, `a11y` to generate their reports and write them to disk)